src/ui/component-repository/components/InstallationPlanModal.js

```diff
@@ -0,0 +1,187 @@
+import React, { useState } from 'react';
+import cx from 'classnames';
+import { useSelector, useDispatch } from 'react-redux';
+import { FormattedMessage, injectIntl, intlShape } from 'react-intl';
+import GenericModalContainer from 'ui/common/modal/GenericModalContainer';
+import { Button, Modal, DropdownButton, MenuItem, ButtonGroup } from 'patternfly-react';
+import { Table } from 'react-bootstrap';
+
+import { getInstallPlan, getComponent, getComponentVersion } from 'state/component-repository/components/selectors';
+import { updateInstallPlan, updateAllInstallPlan, installECRComponent, setInstallUninstallProgress } from 'state/component-repository/components/actions';
+import { setVisibleModal } from 'state/modal/actions';
+
+export const MODAL_ID = 'InstallationPlanModal';
+const IGNORED_KEYS = ['version', 'conflictStrategy', 'hasConflicts'];
+const ACTIONS = {
+  SKIP: 'skip',
+  OVERRIDE: 'update',
+  CREATE: 'install',
+};
+
+const InstallationPlanModal = ({
+  intl,
+}) => {
+  const installPlan = useSelector(getInstallPlan);
+  const selectedComponent = useSelector(getComponent);
+  const version = useSelector(getComponentVersion);
+  const dispatch = useDispatch();
+  const [filterType, setFilterType] = useState('all');
+  const [requiredList, setRequiredList] = useState([]);
+
+  const componentList = [];
+
+  if (installPlan) {
+    Object.keys(installPlan)
+      .filter(key => !IGNORED_KEYS.includes(key))
```

Comment:
why not using `reduce` here, like you did in the reducer file?

Reply:
in the reducer file, I'm keeping the object structure, and applying an update, here I'm converting the two dimension object into a one dimension array. I didn't thought on using reduce for that but I think the code might be bigger, let me try.

Reply:
I was able to change the implementation to use reduce and map and kept the same amount of line 👍